src/components/Button/ButtonBase.test.js

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable flowtype/require-valid-file-annotation */
+import React from 'react';
+import { shallow, mount } from 'enzyme';
+import ButtonBase, {
+  XSmallButton,
+  SmallButton,
+  MediumButton,
+  LargeButton,
+} from './ButtonBase';
+
+const sizes = ['xsmall', 'small', 'medium', 'large'];
+const sizeComponents = {
+  xsmall: XSmallButton,
+  small: SmallButton,
+  medium: MediumButton,
+  large: LargeButton,
+};
+
+describe('ButtonBase component', () => {
+  describe('Button with size prop', () => {
+    const wrapper = shallow(<ButtonBase />);
+    const instance = wrapper.instance();
+
+    for (const size of sizes) {
+      it(`should return sized button component for size = ${size}`, () => {
+        const component = instance.getButtonElem(size);
+        expect(component).toBe(sizeComponents[size]);
+      });
+      it(`should render button for ${size} size`, () => {
+        // Using mount so the styles are available in the snapshot
+        const wrapperSizedButton = mount(<ButtonBase size={size} />);
```

Comment:
I was reading up on testing with Enzyme a bit more and do we need to call `unmount` for cleanup? Or is it not necessary in these cases?

Reply:
I think `.unmount()` is only needed if you're trying to test lifecycle hooks. E.g. if you'd like to test if `componentWillUnmount` is called you could use `wrapper.unmount()` to trigger the unmount. 
As mentioned [here](https://airbnb.io/enzyme/docs/api/ReactWrapper/unmount.html).

The garbage collector should clean the memory from unused objects.